From 8c35b9a2e024ea32f321bd35f27b04831b17240b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Roth?= <neolynx@gmail.com>
Date: Mon, 13 Jan 2014 21:16:21 +0100
Subject: [PATCH 26/37] libdvbv5: fix lost ressource in atsc_eit

if the title of the EIT event is longer than the available data,
make sure the allocated buffer is not lost
---
 lib/libdvbv5/descriptors/atsc_eit.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/libdvbv5/descriptors/atsc_eit.c b/lib/libdvbv5/descriptors/atsc_eit.c
index 86a7b11..c42c8e1 100644
--- a/lib/libdvbv5/descriptors/atsc_eit.c
+++ b/lib/libdvbv5/descriptors/atsc_eit.c
@@ -74,6 +74,11 @@ void atsc_table_eit_init(struct dvb_v5_fe_parms *parms, const uint8_t *buf, ssiz
                 atsc_time(event->start_time, &event->start);
 		event->source_id = eit->header.id;
 
+		if(!*head)
+			*head = event;
+		if(last)
+			last->next = event;
+
 		size = event->title_length - 1;
 		if (p + size > endbuf) {
 			dvb_logerr("%s: short read %zd/%zd bytes", __func__,
@@ -83,11 +88,6 @@ void atsc_table_eit_init(struct dvb_v5_fe_parms *parms, const uint8_t *buf, ssiz
                 /* TODO: parse title */
                 p += size;
 
-		if(!*head)
-			*head = event;
-		if(last)
-			last->next = event;
-
 		/* get the descriptors for each program */
 		size = sizeof(union atsc_table_eit_desc_length);
 		if (p + size > endbuf) {
-- 
1.8.3.2

